source/loader/ur_adapter_registry.hpp

```diff
@@ -88,9 +83,56 @@ class AdapterRegistry {
     static constexpr std::array<const char *, 1> knownPlatformNames{
         MAKE_LIBRARY_NAME("ur_adapter_level_zero", "0")};
 
+    template <typename T, typename = std::enable_if_t<
+                              std::is_constructible_v<T, const std::string &>>>
+    std::vector<T> getEnvVarValuesToVec(const std::string envVar,
```

Comment:
I think we already have a function for parsing env var to a vector in ur_utils.hpp: [getenv_to_vec](https://github.com/oneapi-src/unified-runtime/blob/main/source/common/ur_util.hpp#L132)

You could just map that vector of strings to vector of fs::path by doing something like: 
```
std::vector<fs::path> paths;
std::transform(s.cbegin(), s.cend(), std::back_inserter(paths), [](const std::string &s) { return fs::path(s); });
```

Reply:
Done.